libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -418,6 +418,9 @@ - (SFSyncState*)syncUpWithTarget:(SFSyncUpTarget *)target
                          options:(SFSyncOptions *)options
                         soupName:(NSString *)soupName
                      updateBlock:(SFSyncSyncManagerUpdateBlock)updateBlock {
+    if (target == nil) {
```

Comment:
My earlier PR (https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/1892) is indeed the cause of the regression. I'll have a PR to fix the issue shortly.